GliaWidgets/Sources/CoreSDKClient/CoreSDKClient.Interface.swift

```diff
@@ -423,3 +427,37 @@ extension CoreSdkClient.Engagement {
         Self.isTransferredSecureConversation(self)
     }
 }
+
+extension CoreSdkClient {
+    struct NetworkConnectionMonitor {
+        var networkStream: (Bool) -> AsyncStream<NetworkStatus>
+    }
+}
+
+typealias NetworkConnectionMonitor = CoreSdkClient.NetworkConnectionMonitor
+
+extension CoreSdkClient.NetworkConnectionMonitor {
+    static let live: Self = .init(
+        networkStream: { replay in
+            GliaCore.sharedInstance.networkConnectionMonitor.networkStream(replay: replay)
+        }
+    )
+
+    struct Key: DependencyKey {
```

Comment:
I'm still learning the new dependency injection ins and outs. Why is the key declared in Widgets and not in Core, the same way OpenTelemetry was?

Reply:
Good question. We declared the DependencyKey for OpenTelemetry in Core, because we use its single instance in both CoreSDK and WidgetSDK. In this case we have two separate types of NetworkConnectionMonitor: the one in CoreSDK is a class, the one in Widgets is a struct belonging to CoreSdkClient which is just a bridge between CoreSDK and WidgetsSDK. Since we use CoreSdkClient.NetworkConnectionMonitor in WidgetsSDK only, we don't need it to be declared in Core.